src/routes/safe/components/Transactions/helpers/TxParametersDetail/index.tsx

```diff
@@ -53,19 +54,23 @@ export const TxParametersDetail = ({
   isTransactionExecution,
   isOffChainSignature,
 }: Props): ReactElement | null => {
-  const { nonce } = useSelector(currentSafe)
   const threshold = useSelector(currentSafeThreshold) || 1
   const defaultParameterStatus = isOffChainSignature && threshold > 1 ? 'ETH_HIDDEN' : 'ENABLED'
 
+  const [isAccordionExpanded, setIsAccordionExpanded] = useState(false)
+
   const { safeNonce = '' } = txParameters
-  const isSafeNonceFuture = parseInt(safeNonce, 10) > nonce
-  const [isAccordionExpanded, setIsAccordionExpanded] = useState(isSafeNonceFuture)
+  const safeNonceNumber = parseInt(safeNonce, 10)
+  const { lastTxNonce } = useLastTxNonce()
+
+  const isTxNonceInOrder = lastTxNonce ? safeNonceNumber === lastTxNonce + 1 : false
 
   useEffect(() => {
-    if (parseInt(safeNonce, 10) > nonce) {
+    if (lastTxNonce === undefined || Number.isNaN(safeNonceNumber)) return
+    if (safeNonceNumber !== lastTxNonce + 1) {
       setIsAccordionExpanded(true)
     }
-  }, [nonce, safeNonce, txParameters])
+  }, [lastTxNonce, safeNonceNumber])
```

Comment:
Can't you just use `isTxNonceInOrder` here?
```suggestion
  useEffect(() => {
    if (! isTxNonceInOrder) {
      setIsAccordionExpanded(true)
    }
  }, [isTxNonceInOrder])
```

Reply:
The component renders first `safeNonce` empty. The `useEffect` runs when it changes value. As per suggestion the component would always render the Accordion expanded.